controllers/factory/expansion.go

```diff
@@ -66,19 +66,20 @@ func wasCreatedSTS(ctx context.Context, rclient client.Client, pvcName string, n
 		if actualPVC == nil || newPVC == nil {
 			return false
 		}
+
+		if i := newPVC.Spec.Resources.Requests.Storage().Cmp(*actualPVC.Spec.Resources.Requests.Storage()); i != 0 {
+			log.Info("must re-recreate sts, its pvc claim was changed", "size-diff", i)
```

Comment:
I've realized, that  `Storage().Cmp()` - returns not a diff size, but indicator int - is it the same or not. 

 Will fix it later.

Reply:
Thanks for pointing it out! Maybe I could send another fix later.